### PR TITLE
Warnings when building with gcc

### LINKED ISSUE #172: Fix gcc compiler warnings
Update to gcc (pr #155 ) results in 130 warnings about source code.  Note: TI RTOS warnings to be fixed separately in #211 .

### PR DESCRIPTION
**Pull request by jspectacular**
## Summary
There are 130 warnings when building for the embedded controller with gcc in Linux.  Some of these could be serious, e.g. segmentation faults.  This PR fixes 120 warnings.  9 remaining warnings are due to TI RTOS usage and will be fixed separately (see #211 ).  1 final warning is due to the source code in SPI_transfer in TI RTOS.

## Test Plan
Conduct all unit tests in hardware.

## Issues
Fixes #172 .

### CODE REVIEW

File: firmware/ec/src/post/post.c [hunk 1]
```diff
@@ -192,6 +192,7 @@ static void post_move_to_next_subsystem()
  **    RETURN TYPE     : None
  **
  *****************************************************************************/
+#if 0
```
Comment (mdlewisfb) [inline firmware/ec/src/post/post.c]: Did you block comment these out because of warnings that they are declared and unused? I think this will lead to confusion in the future, can you either add a clear comment as to why it's block commented out or (my preferred way) add a GCC pragma surrounding these functions to ignore the warning for just these, along with a comment saying these are implemented and may be needed in the future, but aren't used now?

File: firmware/ec/src/devices/uart/UartMon.c [hunk 2]
```diff
@@ -41,11 +41,18 @@ static UART_Handle UartMon_open(UART_Handle handle, UART_Params *params)
         return NULL;
     }
 
+#if 0
     /* This is a driver limitation - we only support blocking mode for now */
     if (params->readDataMode != UART_MODE_BLOCKING ||
         params->writeDataMode != UART_MODE_BLOCKING) {
         return NULL;
     }
+#else
```
Comment (mdlewisfb) [inline firmware/ec/src/devices/uart/UartMon.c]: Why this change?

File: firmware/ec/src/devices/eth_sw.c [hunk 3]
```diff
@@ -10,14 +10,15 @@
 //*****************************************************************************
 // Standard Header files
 //*****************************************************************************
-#include "inc/devices/eth_sw.h"
-
-#include "inc/devices/88E6071_registers.h"
+#include <driverlib/sysctl.h>
```
Comment (mdlewisfb) [inline firmware/ec/src/devices/eth_sw.c]: Having this first causes issues when I try to compile.
Per the coding guidelines these should be in the following order:
```
   /* Module header[s] */
    …
    /* stdlib headers */
    …
    /* TI includes */
    …
    /* OC includes */
```

With each section alphabetical.

File: firmware/ec/src/devices/eth_sw.c [hunk 4]
```diff
@@ -98,6 +99,7 @@ uint16_t get_interrupt_status(uint8_t port)
 /*****************************************************************************
  * Internal IRQ handler - reads in triggered interrupts and dispatches CBs
  *****************************************************************************/
+#if 0
```
Comment (mdlewisfb) [inline firmware/ec/src/devices/eth_sw.c]: Why was this commented out?

File: firmware/ec/src/devices/ocmp_wrappers/ocmp_eth_sw.c [hunk 5]
```diff
@@ -187,6 +187,7 @@ static ePostCode _probe(void *driver, POSTData *postData)
     return eth_sw_probe(postData);
 }
 
+#if 0
```
Comment (mdlewisfb) [inline firmware/ec/src/devices/ocmp_wrappers/ocmp_eth_sw.c]: Why was this commented out?

File: firmware/ec/src/devices/sbdn9603.c [hunk 6]
```diff
@@ -197,6 +197,7 @@ bool sbd9603_get_regStatus(OBC_Iridium_Status_Data *pIridiumStatusData)
     return ret;
 }
 
+#if 0
```
Comment (mdlewisfb) [inline firmware/ec/src/devices/sbdn9603.c]: Why was this commented out?

File: firmware/ec/src/devices/eth_sw.c (hunk 3 above)
Comment (jspectacular) [reply firmware/ec/src/devices/eth_sw.c]: Yeah, I had this problem when I merged.  It's fixed now.

File: firmware/ec/inc/common/g510.h [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+ */
+
+#ifndef G510_H_
+#define G510_H_
+
+/*****************************************************************************
+ *                               HEADER FILES
+ *****************************************************************************/
+#include "inc/subsystem/testModule/testModule.h"
+#include "common/inc/global/post_frame.h"
```
[the remark this answers is not in the document]
Comment (jspectacular) [reply firmware/ec/inc/common/g510.h]: ok

File: firmware/ec/inc/common/g510.h [hunk 8]
```diff
@@ -0,0 +1,33 @@
+/**
```
[the remark this answers is not in the document]
Comment (jspectacular) [reply firmware/ec/inc/common/g510.h]: ok

File: firmware/ec/inc/common/sbdn9603.h [hunk 9]
```diff
@@ -0,0 +1,30 @@
+/**
```
[the remark this answers is not in the document]
Comment (jspectacular) [reply firmware/ec/inc/common/sbdn9603.h]: ok

File: firmware/ec/src/devices/eth_sw.c (hunk 4 above)
Comment (jspectacular) [reply firmware/ec/src/devices/eth_sw.c]: The function is not used, therefore we get a warning about an unused function.  It seems like it could be useful at some future time, which is why I #if 0'd it.

File: firmware/ec/src/devices/ocmp_wrappers/ocmp_eth_sw.c (hunk 5 above)
Comment (jspectacular) [reply firmware/ec/src/devices/ocmp_wrappers/ocmp_eth_sw.c]: Same reason as line 102 in eth_sw.c

File: firmware/ec/src/devices/sbdn9603.c (hunk 6 above)
Comment (jspectacular) [reply firmware/ec/src/devices/sbdn9603.c]: Same reason as line 102 in eth_sw.c

File: firmware/ec/src/devices/uart/UartMon.c (hunk 2 above)
Comment (jspectacular) [reply firmware/ec/src/devices/uart/UartMon.c]: We get a warning because readDataMode and UART_MODE_BLOCKING are different types.  Both are enums with values of 0 and 1.  Logically, this is the same as just testing if either readDataMode or writeDataMode is true.  If the change is OK, I'd rather remove the #if 0 part.  Thoughts?

File: firmware/ec/src/post/post.c (hunk 1 above)
Comment (jspectacular) [reply firmware/ec/src/post/post.c]: Same reason as line 102 in eth_sw.c.  We can either add a pragma around unused functions, or add the prototype to a header file.  Which is preferred?

File: firmware/ec/inc/utils/ocmp_util.h [hunk 10]
```diff
@@ -38,10 +37,12 @@ OCMPMessageFrame *OCMP_mallocFrame(uint16_t len);
  **    RETURN TYPE     : OCMPMessageFrame
  **
  *****************************************************************************/
-OCMPMessageFrame *
-create_ocmp_msg_frame(OCMPSubsystem subSystem, OCMPMsgType msgtype,
-                      OCMPActionType actionType, uint8_t componentId,
-                      uint16_t parameters, uint8_t payloadSize);
+OCMPMessageFrame *create_ocmp_msg_frame(OCMPSubsystem subSystem,
```
[the remark this answers is not in the document]
Comment (jspectacular) [reply firmware/ec/inc/utils/ocmp_util.h]: @mdlewisfb, can you answer this?

File: firmware/ec/src/devices/uart/gsm.c [hunk 11]
```diff
@@ -22,7 +22,6 @@
 
 #define TESTMOD_TASK_PRIORITY 2
 #define TESTMOD_TASK_STACK_SIZE 2048
-static Char testmodTaskStack[TESTMOD_TASK_STACK_SIZE];
```
[the remark this answers is not in the document]
Comment (jspectacular) [reply firmware/ec/src/devices/uart/gsm.c]: It's unused in this file.  Is it used elsewhere?

File: firmware/ec/src/filesystem/fs_wrapper.c [hunk 12]
```diff
@@ -39,9 +39,6 @@
 #define READ_SIZE 256
 #define WRITE_SIZE 256
 
-static Queue_Struct fsRxMsg;
```
[the remark this answers is not in the document]
Comment (jspectacular) [reply firmware/ec/src/filesystem/fs_wrapper.c]: It's presently unused, which is why we get the warning.  Should we use a pragma around it?

File: firmware/ec/src/interfaces/UART/uartdma.c [hunk 13]
```diff
@@ -63,8 +63,13 @@ static uint8_t ui8uartdmaRxBuf[UART_RXBUF_SIZE];
...
  *****************************************************************************/
-#pragma DATA_ALIGN(pui8ControlTable, 1024)
-
+#if defined(__TI_COMPILER_VERSION__)
+#    pragma DATA_ALIGN(pui8ControlTable, 1024)
+#elif defined(__IAR_SYSTEMS_ICC__)
+#    pragma data_alignment = 1024
+#elif defined(__GNUC__)
+__attribute__((aligned(1024)))
+#endif
```
Comment (jspectacular) [inline firmware/ec/src/interfaces/UART/uartdma.c]: @JoshuaJeyaraj , is this OK?

File: firmware/ec/src/devices/eth_sw.c (hunk 4 above)
Comment (jspectacular) [reply firmware/ec/src/devices/eth_sw.c]: The function is not used, therefore we get a warning about an unused function. It seems like it could be useful at some future time, which is why I #if 0'd it.

File: firmware/ec/src/devices/sbdn9603.c (hunk 6 above)
Comment (jspectacular) [reply firmware/ec/src/devices/sbdn9603.c]: Same reason as line 102 in eth_sw.c

File: firmware/ec/src/devices/uart/gsm.c (hunk 11 above)
[the remark this answers is not in the document]
Comment (JoshuaJeyaraj) [reply firmware/ec/src/devices/uart/gsm.c]: Ah nevermind , that is referencing a different variable with the same name . You can ignore this comment . . .

File: firmware/ec/src/filesystem/fs_wrapper.c (hunk 12 above)
[the remark this answers is not in the document]
Comment (JoshuaJeyaraj) [reply firmware/ec/src/filesystem/fs_wrapper.c]: i had a discussion with the team here , we decided to go ahead with your changes of removing the these unused variables . This might create merge conflicts when alert implementation  is getting merged , we will take care of it then .

File: firmware/ec/src/post/post.c (hunk 1 above)
Comment (jspectacular) [reply firmware/ec/src/post/post.c]: Added comment to clarify intention.

File: firmware/ec/src/devices/eth_sw.c (hunk 4 above)
Comment (jspectacular) [reply firmware/ec/src/devices/eth_sw.c]: Added comment to clarify intention

File: firmware/ec/src/devices/sbdn9603.c (hunk 6 above)
Comment (jspectacular) [reply firmware/ec/src/devices/sbdn9603.c]: Added comment to clarify intention

File: firmware/ec/inc/utils/ocmp_util.h (hunk 10 above)
[the remark this answers is not in the document]
Comment (jspectacular) [reply firmware/ec/inc/utils/ocmp_util.h]: Ran lint and checked in linted files.

File: firmware/ec/src/devices/ocmp_wrappers/ocmp_eth_sw.c (hunk 5 above)
Comment (jspectacular) [reply firmware/ec/src/devices/ocmp_wrappers/ocmp_eth_sw.c]: Added comment to clarify intention

File: firmware/ec/src/devices/uart/UartMon.c (hunk 2 above)
Comment (jspectacular) [reply firmware/ec/src/devices/uart/UartMon.c]: Left code as was and used pragma instead.  Warning is now gone.

File: firmware/ec/src/interfaces/UART/uartdma.c (hunk 13 above)
Comment (JoshuaJeyaraj) [reply firmware/ec/src/interfaces/UART/uartdma.c]: Looks good to me

File: firmware/ec/inc/subsystem/gpp/ebmp.h [hunk 14]
```diff
@@ -53,6 +53,6 @@ typedef enum {
 /*****************************************************************************
  *                           FUNCTION DECLARATIONS
  *****************************************************************************/
-void ebmp_init(Gpp_gpioCfg *driver);
+int ebmp_init(Gpp_gpioCfg *driver);
```
[the remark this answers is not in the document]
Comment (JoshuaJeyaraj) [reply firmware/ec/inc/subsystem/gpp/ebmp.h]: Ah ok , Lets have the return type as ReturnStatus then . . .

[the remark this answers is not in the document]
Comment (jspectacular) [reply firmware/ec/inc/subsystem/gpp/ebmp.h]: OK
